operator/pkg/helmreconciler/common.go

```diff
@@ -116,33 +114,6 @@ func buildInstallTreeString(componentName name.ComponentName, prefix string, sb
 	}
 }
 
-// applyOverlay applies an overlay using JSON patch strategy over the current Object in place.
-func applyOverlay(current, overlay *unstructured.Unstructured) error {
```

Comment:
I am convinced for the installed-state IstioOperator we should remove this. I am not so sure it is correct for *all* purposes. especially if we are doing SSA or this for other types.

Should we do this only for the IstioOperator?

Reply:
Makes sense. I have added a condition to only do this for `IstioOperator` and also added test cases to test field removals.